src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Tasks/DoNotCreateTaskCompletionSourceWithWrongArguments.cs

```diff
@@ -41,19 +41,37 @@ public override void Initialize(AnalysisContext context)
 
                     compilationContext.RegisterOperationAction(operationContext =>
                     {
-                        // Warn if this is `new TCS(object)` with an expression of type `TaskContinuationOptions` as the argument.
-                        var objectCreation = (IObjectCreationOperation)operationContext.Operation;
-                        if ((objectCreation.Type.OriginalDefinition.Equals(tcsGenericType) || (tcsType != null && objectCreation.Type.OriginalDefinition.Equals(tcsType))) &&
-                            objectCreation.Constructor.Parameters.Length == 1 &&
-                            objectCreation.Constructor.Parameters[0].Type.SpecialType == SpecialType.System_Object &&
-                            objectCreation.Arguments.Length == 1 &&
-                            objectCreation.Arguments[0].Value is IConversionOperation conversionOperation &&
-                            conversionOperation.Operand.Type != null &&
-                            conversionOperation.Operand.Type.Equals(taskContinutationOptionsType))
+                        IConversionOperation? conversionOperation = null;
+                        switch (operationContext.Operation.Kind)
+                        {
+                            case OperationKind.ObjectCreation:
+                                // `new TCS(object)` with an expression of type `TaskContinuationOptions` as the argument
+                                var objectCreation = (IObjectCreationOperation)operationContext.Operation;
+                                conversionOperation = MatchInvalidContinuationOptions(objectCreation.Constructor, objectCreation.Arguments);
+                                break;
+
+                            case OperationKind.Invocation:
+                                // `base(object)` to TCS with an expression of type `TaskContinuationOptions` as the argument
+                                var invocation = (IInvocationOperation)operationContext.Operation;
+                                conversionOperation = MatchInvalidContinuationOptions(invocation.TargetMethod, invocation.Arguments);
+                                break;
+                        }
+
+                        if (conversionOperation is not null)
                         {
                             operationContext.ReportDiagnostic(conversionOperation.CreateDiagnostic(Rule));
                         }
-                    }, OperationKind.ObjectCreation);
+                    }, OperationKind.ObjectCreation, OperationKind.Invocation);
+
+                    IConversionOperation? MatchInvalidContinuationOptions(IMethodSymbol targetMethod, ImmutableArray<IArgumentOperation> arguments) =>
+                        (targetMethod.ContainingType.OriginalDefinition.Equals(tcsGenericType) || (tcsType != null && targetMethod.ContainingType.OriginalDefinition.Equals(tcsType))) &&
```

Comment:
nit: I see this is existing before the change, but null check seems redundant here. `Equals` call should already return false.